code/pages/Site.php

```diff
@@ -197,17 +197,21 @@ public function onAfterWrite() {
 	public function requireDefaultRecords() {
 		parent::requireDefaultRecords();
 
-		if(!Site::get()->count()) {
-			$site = new Site();
-			$site->Title = _t('Multisites.DEFAULTSITE', 'Default Site');
-			$site->IsDefault = true;
-			$site->write();
+		if(DB::query("SELECT COUNT(*) FROM \"SiteTree\" WHERE \"ClassName\" = 'Site'")->value() > 0) {
```

Comment:
Nope, literally a slight modification of what the core in SiteTree to detect count:
`DB::query("SELECT COUNT(*) FROM \"SiteTree\"")->value() `